protocols/bftcosi/bftcosi.go

```diff
@@ -97,23 +102,46 @@ type ProtocolBFTCoSi struct {
 
 	// finale signature that this BFTCoSi round has produced
 	finalSignature *BFTSignature
+
+	// our index in the Roster list
+	index int
+	// prepareSignature is the signature generated during the prepare phase
+	// This signature is adapted according to the exceptions that occured during
+	// the prepare phase.
+	prepareSignature []byte
+	// commitSignature is the signature generated during the commit phase
+	// This signature is adapted according to the exceptions that occured during
+	// the commit phase.
+	commitSignature []byte
```

Comment:
If I understood it correctly the above new lines are only there to have some sort of exception handling. Maybe that's overkill but couldn't this fields be bundled into some struct?